### PR TITLE
feat: support for 'watch_videos' endpoint

### LINKED ISSUE #3021: Add support for youtube.com/watch_videos?video_ids=...
### Describe the feature

The URL https://www.youtube.com/watch_videos?video_ids=iOw6RumXuYY,tDVbLgvjIgM will play the videos with the listed IDs, in order.

It would be great if Piped supported the watch_videos endpoint too :)

### Why would this be useful to add?

The watch_videos endpoint is a handy way to send a short list of videos to a friend without the bother of logging in and making yet another playlist that you'll probably never use again.

### Concept(s)

_No response_

### Additional context

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by Bnyro**
closes #3021

### CODE REVIEW

File: src/components/WatchVideo.vue [hunk 1]
```diff
@@ -687,15 +693,23 @@ export default {
         },
         navigateNext() {
             const params = this.$route.query;
-            let url = this.playlist?.relatedStreams?.[this.index]?.url ?? this.video.relatedStreams[0].url;
+            let url;
+            if (this.playlist) {
+                url = this.playlist?.relatedStreams?.[this.index]?.url ?? this.video.relatedStreams[0].url;
+            } else if (this.$route.query.video_ids) {
+                url = this.$route.path + `?index=${this.index + 1}`;
```
Comment (FireMasterK) [inline src/components/WatchVideo.vue:700]: The entire string could be inside the backtick :thinking:

File: src/components/WatchVideo.vue [hunk 2]
```diff
@@ -648,6 +648,12 @@ export default {
             }
         },
         getVideoId() {
+            if (this.$route.query.video_ids) {
+                const videos_list = this.$route.query.video_ids.split(",");
+                this.index = Number(this.$route.query.index ?? 0);
+                return videos_list[this.index];
+            }
```
Comment (FireMasterK) [inline src/components/WatchVideo.vue:655]: This can be undefined if the index is past the video_ids size, we should do some bounds checking below.

Comment (Bnyro) [reply src/components/WatchVideo.vue:655]: Good catch.

**Review by FireMasterK — APPROVED** [review]

**Comment by FireMasterK** [comment]
Thank you!

**Comment by CalAlaera** [comment]
Wow, that was fast! Thanks so much for this feature 👍🏼

**Pull request merged**